### PR TITLE
Support for TextMapPropagator::Fields method

### LINKED ISSUE #762: API to fetch propagation fields set by TextMapPropagator::Inject() method
As per the specs:  https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#fields:

It should be possible to fetch the list of fields set by Inject() method for propagators. 
This is how other SIG implementation support it:
Python: 
https://github.com/open-telemetry/opentelemetry-python/blob/f816287e5ff0e2d18a1001e11fd0ad7e070e3b2e/opentelemetry-api/src/opentelemetry/propagators/textmap.py#L187

dotnet:
https://github.com/open-telemetry/opentelemetry-dotnet/blob/6b7f2dd77cf9d37260a853fcc95f7b77e296065d/src/OpenTelemetry.Api/Context/Propagation/TextMapPropagator.cs#L34

### PR DESCRIPTION
**Pull request by lalitb**
Fixes #762 

## Changes

Adds support for Fields method in TextMapPropagator, and its implementation for b3, w3c, bagger, jaeger, and composite propagator

For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [x] Unit tests have been added
* [x] Changes in public API reviewed

### CODE REVIEW

**Comment by lalitb** [comment]
@open-telemetry/cpp-approvers requesting please review this. It's there for some time.

File: api/include/opentelemetry/baggage/propagation/baggage_propagator.h
Comment (ThomsonTan) [inline api/include/opentelemetry/baggage/propagation/baggage_propagator.h:52]: `return callback(kBaggageHeader)`?

File: api/include/opentelemetry/trace/propagation/b3_propagator.h
Comment (ThomsonTan) [inline api/include/opentelemetry/trace/propagation/b3_propagator.h:156]: `return callback(kB3CombinedHeader)`?

File: api/include/opentelemetry/trace/propagation/http_trace_context.h
Comment (ThomsonTan) [inline api/include/opentelemetry/trace/propagation/http_trace_context.h:166]: `return callback(kBaggageHeader)`?

File: api/include/opentelemetry/trace/propagation/jaeger.h
Comment (ThomsonTan) [inline api/include/opentelemetry/trace/propagation/jaeger.h:61]: `return callback(kJaegerTraceHeader)`?

**Review by ThomsonTan — APPROVED** [review]

File: api/include/opentelemetry/baggage/propagation/baggage_propagator.h
Comment (lalitb) [reply api/include/opentelemetry/baggage/propagation/baggage_propagator.h:52]: Yes, this looks better. Will change the code before merging.

Comment (lalitb) [reply api/include/opentelemetry/baggage/propagation/baggage_propagator.h]: It's done now.

**Pull request merged**
